go/flowctl/cmd-combine.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/estuary/flow/go/bindings"
+	pf "github.com/estuary/protocols/flow"
+	log "github.com/sirupsen/logrus"
+	pb "go.gazette.dev/core/broker/protocol"
+	mbp "go.gazette.dev/core/mainboilerplate"
+)
+
+type cmdCombine struct {
+	Directory   string                `long:"directory" default:"." description:"Build directory"`
+	Source      string                `long:"source" required:"true" description:"Catalog source file or URL to build"`
+	Collection  string                `long:"collection" required:"true" description:"The name of the collection from which to take the schema and key"`
+	Log         mbp.LogConfig         `group:"Logging" namespace:"log" env-namespace:"LOG"`
+	Diagnostics mbp.DiagnosticsConfig `group:"Debug" namespace:"debug" env-namespace:"DEBUG"`
+}
+
+func (cmd cmdCombine) Execute(_ []string) error {
+	defer mbp.InitDiagnosticsAndRecover(cmd.Diagnostics)()
+	mbp.InitLog(cmd.Log)
+
+	log.WithFields(log.Fields{
+		"config":    cmd,
+		"version":   mbp.Version,
+		"buildDate": mbp.BuildDate,
+	}).Info("flowctl configuration")
+	pb.RegisterGRPCDispatcher("local")
+
+	var err error
+	if cmd.Directory, err = filepath.Abs(cmd.Directory); err != nil {
+		return fmt.Errorf("filepath.Abs: %w", err)
+	}
+	var ctx = context.Background()
+
+	catalog, err := buildCatalog(ctx, pf.BuildAPI_Config{
+		CatalogPath: filepath.Join(cmd.Directory, "catalog.db"),
+		Directory:   cmd.Directory,
+		Source:      cmd.Source,
+		SourceType:  pf.ContentType_CATALOG_SPEC,
+	})
+	if err != nil {
+		return err
+	}
+
+	var collection *pf.CollectionSpec
+	for _, c := range catalog.Collections {
+		if c.Collection.String() == cmd.Collection {
+			collection = &c
+			break
+		}
+	}
+	if collection == nil {
+		return fmt.Errorf("The catalog does not define a collection named: %q", cmd.Collection)
+	}
+	var schemaBundle = pf.SchemaBundle{
```

Comment:
This will break if the schema ref's another resource.

The built catalog already has a SchemaBundle loaded for you; https://github.com/estuary/protocols/blob/main/catalog/build_load.go#L44